### PR TITLE
Feature/access annotation by project name

### PR DESCRIPTION
**Pull request by jiemakel**
**What's in the PR**

Further adding to the functionality added in my previous PR (#1595), this one enables accessing the annotation UI also with the project name in addition to project id. 

While the last PR allowed us to access a particular file by its name, we still had to update the project id if/when in our automatic workflow we deleted/recreated the project. After this, we can just keep the project name the same to create a stable URL to a particular document even across project recreations.

**How to test manually**
The parameter is pn. So, to fully use the functionality, point your browser to http://[webanno-server]/annotation.html?#!n=[document_name]&pn=[project_name] instead of http://[webanno-server]/annotation.html?#!d=[document_number]&p=[project_number].

**Automatic testing**
* [ ] PR includes unit tests

**Documentation**
* [ ] PR updates documentation

### CODE REVIEW

**Comment by reckart** [comment]
In the long run, I'd like to switch to an URL scheme which encodes such parameters directly in the URL path, e.g. `http://host.com/webanno/project/12/annotate/23` or http://host.com/webanno/project/sentiment/annotate/document.txt` - maybe also including the user name etc. Not sure yet how this could look exactly.

**Comment by jiemakel** [comment]
This PR also includes path forms for access as did the last one, but at least at that time the path forms didn't work properly due to some interaction with BRAT.

**Comment by reckart** [comment]
There are a number of implications with the parameters embedded in the URL which is why they are currently not being used. E.g. there is afaik no off-the-shelve Wicket-compatible library that would allow interacting conveniently with the browser history API in order to update the URL when switching between documents without having to re-load the entire annotation page.

**Comment by reckart** [comment]
@UWinch could you have a look at this one as well?

**Comment by jiemakel** [comment]
I've updated the branch. 

@reckart There seems to be something odd going on between this repo and my fork, because there is no option anywhere to allow edits from maintainers. The option mentioned in https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests doesn't appear. You wouldn't happen to know what could be amiss here?

**Comment by jiemakel** [comment]
Answering my own question, https://github.community/t/how-can-we-enable-allow-edits-from-maintainers-by-default/2847 notes that this just isn't currently possible when the fork is owned by an organization instead of a personal account. Which is pretty frustrating, but maybe next time I'll reroute the PR through a personal repo to avoid this.

**Comment by reckart** [comment]
@UWinch can you have a look at the PR before it gets stale again?

File: webanno-ui-annotation/src/main/java/de/tudarmstadt/ukp/clarin/webanno/ui/annotation/AnnotationPage.java [hunk 1]
```diff
@@ -110,7 +111,8 @@
  */
 @MountPath(value = "/annotation.html", alt = { "/annotate/${" + PAGE_PARAM_PROJECT_ID + "}",
         "/annotate/${" + PAGE_PARAM_PROJECT_ID + "}/${" + PAGE_PARAM_DOCUMENT_ID + "}",
-        "/annotate-by-name/${" + PAGE_PARAM_PROJECT_ID + "}/${" + PAGE_PARAM_DOCUMENT_NAME + "}" })
+        "/annotate-by-name/${" + PAGE_PARAM_PROJECT_ID + "}/${" + PAGE_PARAM_DOCUMENT_NAME + "}",
+        "/annotate-by-project-and-document-name/${" + PAGE_PARAM_PROJECT_ID + "}/${" + PAGE_PARAM_DOCUMENT_NAME + "}" })
```
[the remark this answers is not in the document]
Comment (jiemakel) [reply webanno-ui-annotation/src/main/java/de/tudarmstadt/ukp/clarin/webanno/ui/annotation/AnnotationPage.java]: Indeed it should. I've fixed it below.

**Comment by reckart** [comment]
Jenkins, can you test this please?

**Pull request merged**